Fix the Oppia Terms of Service website url to show as a link and clickable

Uticodes opened the pull request:
<!-- READ ME FIRST: Please fill in the explanation section below and check off every point from the Essential Checklist! -->
## Explanation
Fix #4760: In Terms of Service the website should be displayed as link and should be clickable

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
<!-- Delete these section if this PR does not include UI-related changes. -->
If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

| Screenshot Before |  Screenshot After
|:-------------------------:|:-------------------------:|
| ![ts](https://user-images.githubusercontent.com/43546652/227206666-2d85fb06-d315-4020-9c96-27ff2fd2893f.jpeg)| ![oppia_terms_of_service_click](https://user-images.githubusercontent.com/43546652/233938402-f6652095-f461-473f-a0c8-6b53d5bff37f.gif)

translatewiki committed:
Localisation updates from https://translatewiki.net. (#4890)

Translation updates

MohitGupta121 committed:
Replace code owner @rt4914 with @BenHenning in CODEOWNERS file. (#4902)

Replace code owner @rt4914 with @BenHenning in CODEOWNERS file.

gp201 committed:
Fix #4622: Android Wiki migration (#4893)

<!-- READ ME FIRST: Please fill in the explanation section below and
check off every point from the Essential Checklist! -->
## Explanation
<!--
- Explain what your PR does. If this PR fixes an existing bug, please
include
- "Fixes #bugnum:" in the explanation so that GitHub can auto-close the
issue
  - when this PR is merged.
  -->

- Fixes #4622
- The PR does the following: This PR moves the android wiki into the
oppia-andoird codebase allowing contributors to make PRs to the wiki
along with changes to the codebase itself. This PR also introduces a
GitHub Action (Runs when a PR is merged into develop and when a user
makes change to the wiki via the web interface) that copies the content
of the wiki folder along with the git history of the wiki folder and
deploys it to the oppia-android wiki repository.

The steps followed to achieve this wiki migration are present here: 
[Oppia Android Wiki Migration V2
(1).pdf](https://github.com/oppia/oppia-android/files/10921370/Oppia.Android.Wiki.Migration.V2.1.pdf)

### Note: Please ensure GitHub actions have write permissions. Please
follow these instructions to enable it:

[GrantingReadandWritePermissionsonGitHub_PDF.pdf](https://github.com/oppia/oppia-android/files/10919834/GrantingReadandWritePermissionsonGitHub_PDF.pdf)


## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If
this PR fixes part of an issue, prefix the title with "Fix part of
#bugnum: ...".)
- [x] Any changes to
[scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets)
files have their rationale included in the PR explanation.
- [x] The PR follows the [style
guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android
Studio
([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and
is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers
([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
<!-- Delete these section if this PR does not include UI-related
changes. -->
If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of
the before & after UI changes
- For the screenshots above, include both English and pseudo-localized
(RTL) screenshots (see [RTL
guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see
[accessibility
guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests
locally & that they're passing

---------

Co-authored-by: Rajat Talesra <rajattalesra4914@gmail.com>
Co-authored-by: Ben Henning <henning.benmax@gmail.com>
Co-authored-by: Akshay Nandwana <akshaynandwana001@gmail.com>
Co-authored-by: Srikanth K B <srikanth.kadaba@gmail.com>
Co-authored-by: Jan <fsharpasharp@users.noreply.github.com>
Co-authored-by: Abhay Garg <54636525+prayutsu@users.noreply.github.com>
Co-authored-by: Arjun Gupta <arjupta.90@gmail.com>
Co-authored-by: Sparsh Agrawal <55937724+Sparsh1212@users.noreply.github.com>
Co-authored-by: Farees Hussain <fareezzhussain@gmail.com>
Co-authored-by: Sarthak Agarwal <agarwal.sarthak262012@gmail.com>
Co-authored-by: Sean Lip <sean@seanlip.org>
Co-authored-by: Apoorv Srivastava <53645584+MaskedCarrot@users.noreply.github.com>
Co-authored-by: Vinita Murthi <murthi.vinita@gmail.com>
Co-authored-by: Ankita Saxena <ankitasonu24@gmail.com>
Co-authored-by: Vojtěch Jelínek <vojtech.jelinek@hey.com>
Co-authored-by: Ben Henning <ben@oppia.org>

translatewiki committed:
Localisation updates from https://translatewiki.net. (#4905)

Translation updates

MohitGupta121 committed:
Fix #4845: Dark mode support to Equations Text (#4866)

<!-- READ ME FIRST: Please fill in the explanation section below and
check off every point from the Essential Checklist! -->
## Explanation

Fixes #4845 : Dark mode support to Equations Text

<!--
- Explain what your PR does. If this PR fixes an existing bug, please
include
- "Fixes #bugnum:" in the explanation so that GitHub can auto-close the
issue
  - when this PR is merged.
  -->

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If
this PR fixes part of an issue, prefix the title with "Fix part of
#bugnum: ...".)
- [x] Any changes to
[scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets)
files have their rationale included in the PR explanation.
- [x] The PR follows the [style
guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android
Studio
([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and
is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers
([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

### Equation Text Screenshots

<img
src="https://user-images.githubusercontent.com/76530270/216761443-21a973e8-20f9-4fbb-a19d-595530dfebb9.jpg"
height="400" style="max-width: 100%"> <img
src="https://user-images.githubusercontent.com/76530270/216761463-b4a49932-7b9c-4b2a-b3e6-96a5a693f058.jpg"
height="400" style="max-width: 100%">


### Equation Text Screen Recording


https://user-images.githubusercontent.com/76530270/216761782-027744b7-a0df-41ca-8547-2e7f52eaea0c.mp4



## Equation Color Test

#### Day Mode Test


![image](https://user-images.githubusercontent.com/76530270/222653431-05f7e726-c8f0-4677-a48e-8ca37c5ea8e4.png)

#### Night Mode Test


![image](https://user-images.githubusercontent.com/76530270/222653513-458e25ce-9508-480e-bdfc-a4ede5e8a0ed.png)


<!-- Delete these section if this PR does not include UI-related
changes. -->
If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of
the before & after UI changes
- For the screenshots above, include both English and pseudo-localized
(RTL) screenshots (see [RTL
guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see
[accessibility
guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests
locally & that they're passing

---------

Co-authored-by: Ben Henning <ben@oppia.org>

MohitGupta121 committed:
[Android Wiki] Fixed Typo error in Bazel Installation wiki page. (#4911)

<!-- READ ME FIRST: Please fill in the explanation section below and
check off every point from the Essential Checklist! -->
## Explanation

[Android Wiki] Fixed Typo error in Bazel Installation wiki page.

<!--
- Explain what your PR does. If this PR fixes an existing bug, please
include
- "Fixes #bugnum:" in the explanation so that GitHub can auto-close the
issue
  - when this PR is merged.
  -->

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If
this PR fixes part of an issue, prefix the title with "Fix part of
#bugnum: ...".)
- [x] Any changes to
[scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets)
files have their rationale included in the PR explanation.
- [x] The PR follows the [style
guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android
Studio
([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and
is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers
([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

Co-authored-by: Ben Henning <ben@oppia.org>

MohitGupta121 committed:
Fixed #4907 : [Android Wiki] "Get-Help" - Add link for Debugging Docs guide  (#4908)

<!-- READ ME FIRST: Please fill in the explanation section below and
check off every point from the Essential Checklist! -->
## Explanation

Fixed #4907 : [Android Wiki] "Get-Help" - Add link for Debugging Docs
guide

<!--
- Explain what your PR does. If this PR fixes an existing bug, please
include
- "Fixes #bugnum:" in the explanation so that GitHub can auto-close the
issue
  - when this PR is merged.
  -->

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If
this PR fixes part of an issue, prefix the title with "Fix part of
#bugnum: ...".)
- [x] Any changes to
[scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets)
files have their rationale included in the PR explanation.
- [x] The PR follows the [style
guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android
Studio
([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and
is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers
([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

Uticodes committed:
I made the Oppia Terms of Service website url to show as a link and clickable

kkmurerwa committed:
Fix part of #3602: Add accessibility label for ProfileEditActivity (#4894)

<!-- READ ME FIRST: Please fill in the explanation section below and
check off every point from the Essential Checklist! -->
## Explanation
<!--
- Explain what your PR does. If this PR fixes an existing bug, please
include
- "Fixes #bugnum:" in the explanation so that GitHub can auto-close the
issue
  - when this PR is merged.
  -->
Fix part of #3602: When merged, this PR will;
- Create a strings item in vaues/strings.xml with the accessibility text
label for the ProfileEditActivity.
- Add a label for the activity in the manifest file.
- Remove the activity from the accessibility_label_exemptions.textproto
file.
- Add test to check that the appropriate label is displayed for
ProfileEditActivity.
- Modified tests in ProfileEditActivityTest to reflect new title
behavior and remove any checks for the username as the toolbar title.
- Removed deprecated testProfileEdit_updateName_checkNewNameDisplayed()
test that checked for previous title behaviour.

New changes after 22/03/23 review;
- Rename the string from R.string.profile_edit_activity_label to
R.string.profile_edit_activity_title
- Update the Manifest file to reflect this change.
- Modify the ProfileEditActivityTest class to reflect the change and
test for the toolbar title.
- Linting check fixes.

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If
this PR fixes part of an issue, prefix the title with "Fix part of
#bugnum: ...".)
- [x] Any changes to
[scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets)
files have their rationale included in the PR explanation.
- [x] The PR follows the [style
guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android
Studio
([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and
is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers
([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

---------

Co-authored-by: Ben Henning <ben@oppia.org>

BenHenning commented:
@Uticodes it seems like this PR has additional extra changes. Please make sure to sync it with the latest develop changes (and make sure that your fork also has the latest develop changes).

seanlip committed:
Fix #248, #501, #179, #762, #207: Remove obsolete TODOs. (#4928)

## Explanation

Fixes #248
Fixes #501
Fixes #179
Fixes #762
Fixes #207

This PR removes some obsolete TODOs so that the corresponding issues on
GitHub can be properly closed.

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If
this PR fixes part of an issue, prefix the title with "Fix part of
#bugnum: ...".)
- [x] Any changes to
[scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets)
files have their rationale included in the PR explanation.
- [x] The PR follows the [style
guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android
Studio
([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and
is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers
([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

masclot committed:
Fix part of #632: Move PromotedStoryListAdapter to BindableAdater (#4874)

<!-- READ ME FIRST: Please fill in the explanation section below and
check off every point from the Essential Checklist! -->
## Explanation
<!--
- Explain what your PR does. If this PR fixes an existing bug, please
include
- "Fixes #bugnum:" in the explanation so that GitHub can auto-close the
issue
  - when this PR is merged.
  -->

Fixes part of #632: Reimplementation of PromotedStoryListAdapter as
BindableAdater
Espresso tests are failing, but they were already failing. The errors
with the new adapter are the same. The same tests pass when using
gradle.

There was an error in MarginBindingAdapter which I'm also fixing in this
cl: MarginBindingAdapter was replacing start/end margin changes for
left/right margin changes. However, this replacement relied on getting
the layout direction from the View being changed. When the direction is
inherited and the View is not attached to a parent (recursively until
top level View), the direction defaults to LTR, no matter the
Locale/language. This situation happens for example in item Views within
RecyclerView. The change does not replace start/end any more, but sets
them through MarginLayoutParamsCompat so that it is compatible with api
< 17.

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If
this PR fixes part of an issue, prefix the title with "Fix part of
#bugnum: ...".)
- [X] Any changes to
[scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets)
files have their rationale included in the PR explanation.
- [X] The PR follows the [style
guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [X] The PR does not contain any unnecessary code changes from Android
Studio
([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [X] The PR is made from a branch that's **not** called "develop" and
is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers
([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

| Current | new |
|-|-|
|
![bindable_adapter_current](https://user-images.githubusercontent.com/103062089/217546850-30d59ac3-d996-4856-9a55-43a61e533eb2.png)
|
![bindable_adapter_new](https://user-images.githubusercontent.com/103062089/217546857-224a287d-b55a-4af9-8b05-e4855d78bd44.png)
|

Unit tests:

![recently_played_span_test](https://user-images.githubusercontent.com/103062089/217547368-f86fa7c4-c915-4257-948c-90a8e73c3ab6.png)

Espresso Tests. The run for the new version has an extra error, but it
is due to timings. In other runs it passes.

![recently_played_espresso_tests_current_vs_new](https://user-images.githubusercontent.com/103062089/217548889-21e0599e-e8b0-463d-a3f6-eeaaa93e674a.png)

@rt4914 PTAL

---------

Co-authored-by: Ben Henning <ben@oppia.org>

masclot committed:
Fix #3755: Administration controls invisible but actionable (#4901)

<!-- READ ME FIRST: Please fill in the explanation section below and
check off every point from the Essential Checklist! -->
## Explanation
Fix #3755: AdministratorControlsActivity tablet has Fragment management
issues

The "App Version" fragment was being added on top of the previous
fragment, instead of replacing it.

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If
this PR fixes part of an issue, prefix the title with "Fix part of
#bugnum: ...".)
- [x] Any changes to
[scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets)
files have their rationale included in the PR explanation.
- [x] The PR follows the [style
guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android
Studio
([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and
is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers
([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
<!-- Delete these section if this PR does not include UI-related
changes. -->
If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of
the before & after UI changes
- For the screenshots above, include both English and pseudo-localized
(RTL) screenshots (see [RTL
guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see
[accessibility
guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests
locally & that they're passing


The following videos show how you can click on a hidden ui control in
the current version, but you cannot after the changes in this cl.
|current|new|
|-|-|

|[current_administration_controls_invisible_but_actionable.webm](https://user-images.githubusercontent.com/103062089/225941610-d41ed412-625c-45de-b18c-3279bd2d2401.webm)|[new_administration_controls_invisible_but_actionable.webm](https://user-images.githubusercontent.com/103062089/225941644-7b3598ba-89c5-4a8b-9611-ecc751ea1b25.webm)|

Screenshot with test failure when app-version goes back to use add
fragment, instead of replace fragment.

![Test failure when app-version adds fragment, instead of replacing
it](https://user-images.githubusercontent.com/103062089/228270103-04a9224e-5ca2-4150-9a8e-abc717972290.png)

---------

Co-authored-by: Ben Henning <ben@oppia.org>

MohitGupta121 committed:
[Android Wiki] Added "Tips and Tricks" for android wiki contribution (#4921)

<!-- READ ME FIRST: Please fill in the explanation section below and
check off every point from the Essential Checklist! -->
## Explanation

[Android Wiki] Added "Tips and Tricks" for android wiki contribution 

- Tips for Wiki Markdown Preview in Android Studio 


<!--
- Explain what your PR does. If this PR fixes an existing bug, please
include
- "Fixes #bugnum:" in the explanation so that GitHub can auto-close the
issue
  - when this PR is merged.
  -->

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If
this PR fixes part of an issue, prefix the title with "Fix part of
#bugnum: ...".)
- [x] Any changes to
[scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets)
files have their rationale included in the PR explanation.
- [x] The PR follows the [style
guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android
Studio
([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and
is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers
([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

Uticodes committed:
Merge branch 'develop' into in_terms_of_service_the_website_should_be_displayed_as_link_and_should_be_clickable

Uticodes commented:
@BenHenning PR now in sync with develop, kindly check again.

BenHenning commented:
Thanks @Uticodes! As discussed in #4760 we ideally want a solution here that's a bit more future proof than just changing the XML (since we plan to generate it long-term). One suggestion in the issue thread was to try and use [LinkifyCompat](https://developer.android.com/reference/kotlin/androidx/core/text/util/LinkifyCompat) for this purpose. Could you please try that and see if it linkifies the URL without needing to modify the XML directly?

Uticodes committed:
Merge branch 'develop' into in_terms_of_service_the_website_should_be_displayed_as_link_and_should_be_clickable

Uticodes committed:
I made the Oppia Terms of Service website url to show as a link and clickable

Uticodes committed:
Merge remote-tracking branch 'origin/in_terms_of_service_the_website_should_be_displayed_as_link_and_should_be_clickable' into in_terms_of_service_the_website_should_be_displayed_as_link_and_should_be_clickable

Uticodes committed:
I made the Oppia Terms of Service website url to show as a link and clickable

Uticodes committed:
I made the Oppia Terms of Service website url to show as a link and clickable

Uticodes committed:
trying to fix Imports must be ordered in lexicographic order error

Uticodes committed:
fixed Imports must be ordered in lexicographic order error

adhiamboperes commented on app/src/main/java/org/oppia/android/app/policies/PoliciesFragmentPresenter.kt:
```diff
@@ -78,6 +80,9 @@ class PoliciesFragmentPresenter @Inject constructor(
       supportsConceptCards = false
     )
+
+    val websitePattern = Pattern.compile("https://www.oppia.org")
```
This pattern will only be useful for finding links to this address, but if we would like to find **any** link in the span, we may use something like:
```suggestion
    val websitePattern = Pattern.compile("(http|ftp|https)://([\\w_-]+(?:(?:\\.[\\w_-]+)+))([\\w.,@?^=%&:/~+#-]*[\\w@?^=%&/~+#-])")
```

Uticodes replied:
cool, let me test this and see if it works, thanks @adhiamboperes

adhiamboperes reviewed (changes requested):
Thanks @Uticodes.

Please also update the video in your description to test the ToS page as opposed to the Privacy Policy page, as well as an explanation of the fix.

Uticodes commented:
> @Uticodes it seems like this PR has additional extra changes. Please make sure to sync it with the latest develop changes (and make sure that your fork also has the latest develop changes).

Sure will do, thanks @BenHenning

Uticodes commented:
> Thanks @Uticodes.
> 
> Please also update the video in your description to test the ToS page as opposed to the Privacy Policy page, as well as an explanation of the fix.

sure, on it.

Uticodes committed:
Merge branch 'develop' into in_terms_of_service_the_website_should_be_displayed_as_link_and_should_be_clickable

Uticodes committed:
update website Pattern

Uticodes committed:
fix lint

Uticodes commented:
@BenHenning @adhiamboperes PTAL
I have update branch, PR and the video to show how it works, kindly review again 🙏

adhiamboperes reviewed (commented):
I think this fix works though I think we should try to have an automated test for this.
Can you use one of the other URL tests as a template?

BenHenning commented on app/src/main/res/values/terms_of_service.xml:
```diff
@@ -4,8 +4,8 @@
-      operates the website located at https://www.oppia.org (the "Website"). The
-      following Terms of Use (also referred to as the "Terms") are a contract
+      operates the website located at https://www.oppia.org (the "Website").
+      The following Terms of Use (also referred to as the "Terms") are a contract
```
Is this wrapping actually needed to fix the issue?

BenHenning commented on app/src/main/java/org/oppia/android/app/policies/PoliciesFragmentPresenter.kt:
```diff
@@ -78,6 +80,12 @@ class PoliciesFragmentPresenter @Inject constructor(
+      "(http|ftp|https)://([\\w_-]+(?:(?:\\." +
+        "[\\w_-]+)+))([\\w.,@?^=%&:/~+#-]*[\\w@?^=%&/~+#-])"
+    )
+    LinkifyCompat.addLinks(binding.policyDescriptionTextView, websitePattern, "https://")
```
Is this something we can add generally to our HtmlParser class? E.g. here: https://github.com/oppia/oppia-android/blob/a2182477dcdda96d8eaeb8815d13d96fd4093380/utility/src/main/java/org/oppia/android/util/parser/html/HtmlParser.kt#L110

BenHenning commented on app/src/main/java/org/oppia/android/app/policies/PoliciesFragmentPresenter.kt:
```diff
@@ -78,6 +80,12 @@ class PoliciesFragmentPresenter @Inject constructor(
       supportsConceptCards = false
     )
+
+    val websitePattern = Pattern.compile(
+      "(http|ftp|https)://([\\w_-]+(?:(?:\\." +
+        "[\\w_-]+)+))([\\w.,@?^=%&:/~+#-]*[\\w@?^=%&/~+#-])"
+    )
```
As indicated in my link in the comment below, we currently linkify using Linkify.WEB_URLS. Why isn't that pattern working in our case?

BenHenning reviewed (changes requested):
Thanks @Uticodes! This is definitely on the right track, I think. I left two comments--PTAL.

adhiamboperes committed:
Merge branch 'develop' into in_terms_of_service_the_website_should_be_displayed_as_link_and_should_be_clickable
